### PR TITLE
Add `ConnectivityIntegration` for web

### LINKED ISSUE #1530: Network connection change breadcrumbs for Web
### Description

https://github.com/getsentry/team-mobile/issues/82

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Based on https://github.com/ueman/sentry-dart-tools/tree/main/sentry_connectivity

Adds breadcrumb 'wifi' when online, and breadcrumb 'none' when offline.

## :bulb: Motivation and Context

Closes #1530

## :green_heart: How did you test it?

Tests, run example in web browser.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/pubspec.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ dependencies:
   package_info_plus: '>=1.0.0 <=5.0.1'
   meta: ^1.3.0
   ffi: ^2.0.0
+  connectivity_plus: ^5.0.2
```
Comment (ueman) [inline flutter/pubspec.yaml]: This is kinda against https://develop.sentry.dev/philosophy/#dependencies-cost and as known from package_info_plus it's a quite frequent source of major updates.

Comment (buenaflor) [inline flutter/pubspec.yaml]: Pls check if we can do this without having to add another package

**Comment by denrase** [comment]
@ueman @buenaflor 

I checked and `connectivity_plus` seems to be the go-to package to check network availability on dart/flutter. It's part of the flutter community packages. I looked into the implementations, excluding iOS and Android.

## Implementations

### Web

Thre are (limitiations)[https://pub.dev/packages/connectivity_plus#limitations-on-the-web-platform] on the web platform. The package uses the [NetworkInformation](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation), but it's not widley available. As a fallback we have `navigator.onLine`, but it does not give us information about the network type, only if we are online/offline.

https://github.com/fluttercommunity/plus_plugins/blob/a928f67f7a6232373dc8caa00e80dbf78619fcc6/packages/connectivity_plus/connectivity_plus/lib/src/web/dart_html_connectivity_plugin.dart#L26

### Linux

It's using the [nm](https://pub.dev/packages/nm) package for network connectivity. So again a dependency.

https://github.com/fluttercommunity/plus_plugins/blob/a928f67f7a6232373dc8caa00e80dbf78619fcc6/packages/connectivity_plus/connectivity_plus/lib/src/connectivity_plus_linux.dart#L5

### Windows

CPP implementation based on [Win32NetState](https://github.com/PurpleI2P/i2pd/blob/master/Win32/Win32NetState.cpp) using method channels.

https://github.com/fluttercommunity/plus_plugins/blob/a928f67f7a6232373dc8caa00e80dbf78619fcc6/packages/connectivity_plus/connectivity_plus/windows/connectivity_plus_plugin.cpp

## Conclusion

We'd have to provide an implementation of for every platform:

- **Web**: Fairly straightforward, as we can use `dart:html` for online/offline state.
- **Linux**: The approach used by connectivity_plus still requires a dependency.
- **Windows**: CPP implementation communicating through message channels.

WDYT and do you have other suggestions how to proceed here? If we don't wan't an additional dependency in the main packages, we could provide the integration is a separate package, like we do for the database integrations.

**Comment by ueman** [comment]
I'm guessing web is straight forward to implement (as Flutter for Web anyway requires a pretty new browser) and iOS/macOS/Android already have connectivity breakdcrumbs in their SDKs. So Windows and Linux are the ones without information. So maybe the solution is to add Sentry Native to Windows/Linux versions of Sentry-Flutter if that has the connectivity information?

**Comment by denrase** [comment]
@buenaflor Do we have stats how much usage the SDK has on linux/windows apps? If it's very low might be good enough to do web only for now, as we do have iOS/Android through the native SDKs.

**Comment by buenaflor** [comment]
After discussion we will use this PR to implement web and create separate issues for windows and linux and prio accordingly

**Comment by denrase** [comment]
The `NetworkInformationAPI` is still experimental, so we only check `online/offline` on web.

https://developer.mozilla.org/en-US/docs/Web/API/Network_Information_API

https://github.com/fluttercommunity/plus_plugins/blob/5e5c9db8d640f80ae170f3738489a900328eda75/packages/connectivity_plus/connectivity_plus/lib/src/connectivity_plus_web.dart#L15

**Review by buenaflor — APPROVED** [review]
nice 👍

**Pull request merged**